### PR TITLE
Append extraVolumes instead of force set them

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/roadmap/issues/572

The `api-server` `extraVolumes` on the `KubeadmControlPlane` resource are being set from kyverno. But this is overwriting whatever `extraVolumes` were set in the first place. We have to have another one to mount the cloud configuration for the cloud provider. We are currently [setting that in the `kubectl-gs` template](https://github.com/giantswarm/kubectl-gs/blob/2a0b07b28dbddce25ab11606f57bd76e408258f3/cmd/template/cluster/provider/templates/azure/kubeadm_control_plane.yaml.tmpl#L16-L23), but it gets overwritten by kyverno. This currently breaks cluster creation.

With these changes we append these `extraVolumes` to whatever is already there.

### Checklist

- [X] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by whites11** [comment]
Don't kill me. How about adding unit tests?

**Comment by fiunchinho** [comment]
> Don't kill me. How about adding unit tests?

Yeah, definitely. I'm working on that

**Review by calvix — APPROVED** [review]
lgtm if tested

File: helm/policies-common/templates/CoreCAPI.yaml [hunk 1]
```diff
@@ -261,14 +261,6 @@ spec:
                   +(tls-cipher-suites): "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_GCM_SHA256"
                   +(runtime-config): "api/all=true,scheduling.k8s.io/v1alpha1=true"
                   +(feature-gates): "TTLAfterFinished=true"
-                extraVolumes:
```
Comment (fiunchinho) [inline helm/policies-common/templates/CoreCAPI.yaml:264]: We don't force set the extraVolumes but rather we append them later.

File: helm/tests/ensure.py [hunk 2]
```diff
@@ -742,6 +742,15 @@ def kubeadm_control_plane(kubernetes_cluster):
         spec:
           kubeadmConfigSpec:
             clusterConfiguration:
+              apiServer:
+                extraArgs:
+                  cloud-config: /etc/kubernetes/azure.json
```
Comment (fiunchinho) [inline helm/tests/ensure.py:747]: I'm adding these to the common fixture so we can later assert that we are not overwriting existing configuration.

**Review by njuettner — APPROVED** [review]
lgtm

**Pull request merged**